robot_controllers_interface/src/controller_manager.cpp

```diff
@@ -203,10 +203,33 @@ void ControllerManager::reset()
   }
 }
 
-bool ControllerManager::addJointHandle(JointHandlePtr& j)
+bool ControllerManager::addJointHandle(JointHandlePtr& jointHandlePtr)
 {
-  // TODO: check for duplicate names?
-  joints_.push_back(j);
+  // If we don't already have a handle with
+  // a duplicate name, we add it
+
+  for(auto jointHandlePointer : joints_)
+  {
+    if(jointHandlePointer->getName() == jointHandlePtr->getName())
+    return false;
+  }
+  
+  joints_.push_back(jointHandlePtr);
+  return true;
+}
+
+bool ControllerManager::addGyroHandle(GyroHandlePtr gyroHandlePtr)
+{
+  // If we don't already have a handle with
+  // a duplicate name, we add it
+
+  for(auto gyroHandlePointer : gyros_)
```

Comment:
use underscore for variables.   For local variables don't put trailing underscore.
```
 gyro_handle_pointers
```

Reply:
Also, just a note - spacing doesn't follow the ROS C++ style guide (should be a space after if/for in all of these statements).